api/src/main/java/com/epam/pipeline/manager/cluster/writer/MonitoringStatsWriter.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2017-2020 EPAM Systems, Inc. (https://www.epam.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.manager.cluster.writer;
+
+import com.epam.pipeline.entity.cluster.monitoring.MonitoringStats;
+import com.opencsv.CSVWriter;
+import lombok.NoArgsConstructor;
+import lombok.Value;
+
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@NoArgsConstructor
+public class MonitoringStatsWriter {
+
+    private static final List<String> COMMON_STATS_HEADER =
+        Arrays.asList("Timestamp", "CPU_cores", "CPU_usage[%]", "MEM_capacity[bytes]", "MEM_usage[%]");
+    private static final String DISK_TOTAL_HEADER_TEMPLATE = "%s_total[bytes]";
+    private static final String DISK_USAGE_HEADER_TEMPLATE = "%s_usage[%%]";
+    private static final String NETWORK_USAGE_IN_HEADER_TEMPLATE = "%s_in[bytes]";
+    private static final String NETWORK_USAGE_OUT_HEADER_TEMPLATE = "%s_out[bytes]";
+    private static final double HUNDRED_PERCENTS = 100.0;
+
+    public String convertStatsToCsvString(final List<MonitoringStats> stats) {
+        final StringWriter stringWriter = new StringWriter();
+        final CSVWriter csvWriter = new CSVWriter(stringWriter);
+        final List<String[]> allLines = extractTable(stats);
+        csvWriter.writeAll(allLines);
+        return stringWriter.toString();
+    }
+
+    private List<String[]> extractTable(final List<MonitoringStats> stats) {
+        final List<String[]> allLines = new ArrayList<>();
+        final MonitoringStatsHeader header = extractHeader(stats);
+        if (!header.getColumnNames().isEmpty()) {
```

Comment:
You can wrap collections into `ListUtils.emptyIfNull`

Reply:
Removed during refactoring: now checking if the whole stats list is empty at the beginning of `convertStatsToCsvString`